R/broom_wrapper.R

```diff
@@ -94,8 +94,33 @@ predict <- function(df, model, ...){
 
 #' apply data frame with model to a data frame
 #' @export
-add_prediction <- function(df, model_df){
-  broom::augment(model_df, model, newdata = df)
+add_prediction <- function(df, model_df, ...){
+
+  tryCatch({
+    broom::augment(model_df, model, newdata = df, ...)
+  }, error = function(e){
+    if (grepl("arguments imply differing number of rows: ", e$message)) {
+      # in this case, df has categories that aren't in model
+
+      filtered_data <- df
+
+      for(model in model_df[["model"]]){
```

Comment:
Looks like this is looping over all the columns, but doesn't this do the filtering on numeric columns too?